### DISCUSSION TITLE
Is there a simple alternative to this code

### DISCUSSION THREAD

**Discussion by pclewisnz** [dfinke/ImportExcel#1264] [Q&A] [answered]
```
$excel=(New-Object -TypeName PSObject -Property (@{'id'=23; 'start'="2022-09-01 00:02:50.816"}))|export-excel -FreezeTopRow -path "c:\temp\test2.xlsx" -TableStyle "Light10" -AutoSize -PassThru
$sheet=$excel.Workbook.Worksheets[1]
$sheet.Column($sheet.tables[0].columns["start"].Id)| Set-ExcelRange  -NFormat "dd-MM-yyyy HH:mm:ss.000"
Close-ExcelPackage $excel
```
After exporting, have to set the Number Format for all my datetime fields to include milliseconds
If there are 20 datetime fields then with code above I have to duplicate the Set-ExcelRange line for each 
Alternatively I can pull the name of the columns and likely the position the sheet from the PSCustomObject.
In my use case the data is string and I don't know which fields are datetime so I'd have to test
`"2022-09-01 00:02:50.816"| %{ (get-date $_).ToString("yyyy-MM-dd HH:mm:ss.fff") -eq $_ }`

This is just in case there's a switch with a subtle purpose and name I have not seen in which I can globally set the Number Format for any date/time field in the sheet.  Maybe there's a better way - like overloading one of the modules functions do to the above type detection/conversion (expensive on large exports)?

**Comment by dfinke** [comment, marked as answer]
Created issue https://github.com/dfinke/ImportExcel/issues/1265
